cpp/src/sort/sort_impl.cuh

```diff
@@ -156,6 +160,8 @@ std::unique_ptr<column> sorted_order(table_view input,
                    mutable_indices_view.end<size_type>(),
                    comparator);
     }
+    // protection for temporary d_column_order
+    stream.synchronize();
   }
```

Comment:
```suggestion
}

// protection for temporary d_column_order
stream.synchronize();
```

The `stream.synchronize();` statement is called at the end of each blocks `if else`, thus we can move it out of those blocks.